Neos.ContentRepository.Core/Classes/Feature/Common/TetheredNodeInternals.php

```diff
@@ -133,12 +133,17 @@ protected function createEventsForMissingTetheredNode(
                 );
             }
 
-            $childNodeSource = null;
-            foreach ($childNodeAggregate->getNodes() as $node) {
-                $childNodeSource = $node;
-                break;
+            $rootDimensionSpacePoints = $this->interDimensionalVariationGraph->getRootGeneralizations();
+            if (empty($rootDimensionSpacePoints)) {
+                // safeguard
+                throw new \RuntimeException(
+                    'The dimension space is empty, please check your configuration.',
+                    1706864056
+                );
             }
```

Comment:
todo add non empty array as return type for `getRootGeneralizations`, cause its NEVER empty, its part of the interface